posthog.go

```diff
@@ -245,21 +260,28 @@ func (c *client) ReloadFeatureFlags() error {
 	return nil
 }
 
-func (c *client) GetFeatureFlag(flagKey string, distinctId string, defaultValue interface{}) (interface{}, error) {
+func (c *client) GetFeatureFlag(flagConfig FeatureFlagPayload) (interface{}, error) {
+	if err := flagConfig.validate(); err != nil {
+		return false, err
+	}
+
 	if c.featureFlagsPoller == nil {
```

Comment:
`personalApiKey` is not required for calling `/decide`, only for local eval

Reply:
hmm well here's another consideration to make. the typing is strict here and currently the typing for creating the feature flag object requires a projectapikey. If we want the key to be optional, refactoring needs to happen on how the arguments are accepted

Reply:
Fair enough :)